src/controllers/expense.js

```diff
@@ -0,0 +1,121 @@
+'use strict';
+
+const {
+  getAllExpenses,
+  getExpense,
+  deleteExpense,
+  createNewExpense,
+  updateExpense,
+} = require('../services/expenses');
+
+const getAllExpensesController = (req, res) => {
+  const { userId, category, from, to } = req.query;
+
+  if (!category || !userId || !from || !to) {
+    res.sendStatus(422);
+
+    return;
+  }
+
+  const findExpenses = getAllExpenses(
+    userId,
+    from,
+    to,
+    category
+  );
+
+  if (!findExpenses.length) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(findExpenses);
+};
+
+const getExpenseController = (req, res) => {
+  const { expenseId } = req.params;
+
+  if (!getExpense(expenseId)) {
+    res.sendStatus(404);
+
+    return;
+  }
+
+  res.send(getExpense(expenseId));
+};
+
+const deleteExpenseController = (req, res) => {
+  const { expenseId } = req.params;
+
+  const status = deleteExpense(expenseId);
+
+  if (!status) {
+    res.sendStatus(404);
+  }
+
+  res.sendStatus(204);
+};
+
+const createNewExpenseController = (req, res) => {
+  const { userId, spentAt, title, amount, category, note } = req.body;
+
+  if (!userId
+    || !spentAt
+    || !title
+    || !amount
+    || !category
+    || !note) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const newExpense = createNewExpense(
+    userId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note
+  );
+
+  res.statusCode = 201;
+  res.send(newExpense);
+};
+
+const updateExpenseController = (req, res) => {
+  const { expenseId } = req.params;
+  const { spentAt, title, amount, category, note } = req.body;
+
+  if (!expenseId || !spentAt || !title || !amount || !category || !note) {
+    res.sendStatus(400);
+
+    return;
+  }
+
+  const status = updateExpense(
+    expenseId,
+    spentAt,
+    title,
+    amount,
+    category,
+    note
```

Comment:
Same here, only id is used as separate constant inside `updateExpense` function, you can pass object here